airbyte-integrations/connectors/source-sftp/src/main/java/io/airbyte/integrations/source/sftp/SftpSource.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.source.sftp;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.common.collect.Lists;
+import io.airbyte.commons.util.AutoCloseableIterator;
+import io.airbyte.commons.util.AutoCloseableIterators;
+import io.airbyte.integrations.BaseConnector;
+import io.airbyte.integrations.base.IntegrationRunner;
+import io.airbyte.integrations.base.Source;
+import io.airbyte.protocol.models.*;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class SftpSource extends BaseConnector implements Source {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SftpSource.class);
+
+    public static void main(String[] args) throws Exception {
+        new IntegrationRunner(new SftpSource()).run(args);
+    }
+
+    @Override
+    public AirbyteConnectionStatus check(JsonNode config) throws Exception {
+        final SftpClient client = new SftpClient(config);
+        try {
+            final SftpCommand command = new SftpCommand(client, config);
+            client.connect();
+            String workingDirectory = config.has("folder_path") ? config.get("folder_path").asText() : "";
+            if (StringUtils.isNotBlank(workingDirectory)) {
+                command.tryChangeWorkingDirectory(workingDirectory);
+            }
+            client.disconnect();
+            return new AirbyteConnectionStatus().withStatus(AirbyteConnectionStatus.Status.SUCCEEDED);
+        } catch (Exception e) {
+            LOGGER.error("Exception attempting to connect to the server: ", e);
+            return new AirbyteConnectionStatus()
+                    .withStatus(AirbyteConnectionStatus.Status.FAILED)
+                    .withMessage("Could not connect to the server with provided configuration. \n" + e.getMessage());
+        } finally {
+            client.disconnect();
+        }
+    }
+
+
+    @Override
+    public AirbyteCatalog discover(JsonNode config) throws Exception {
```

Comment:
Could you add a comment about what the implementation of this method does?

Reply:
It is already documented in the parent interface Source 
```
  /**
   * Discover the current schema in the source.
   *
   * @param config - integration-specific configuration object as json. e.g. { "username": "airbyte",
   *        "password": "super secure" }
   * @return Description of the schema.
   * @throws Exception - any exception.
   */
  AirbyteCatalog discover(JsonNode config) throws Exception;
```
  
  But maybe it may sense to add an extra one for Sftp implementation with more sftp discover details